Python doc (#33)
CTSRD-CHERI/chericat: python/README.md

> # Chericat Python

Reviewer: `/README.md` should somehow reference this. Otherwise, I'm afraid people won't read it.

Author: Fixed.

> # Chericat Python

Reviewer: ```suggestion
# chericat graph visualisation
```

Author: Fixed.

A set of python code to display Chericat capabilities related data on graphs

> Packages required and their tested versions to run the programs:

Reviewer: Could you provide a list of commands a user must execute, possibly in a separate section here, to prepare the environment? This should include `pkg64`/`pkg64c` commands that install required packages.

Author: Added a column to the table and made it into a section. Does it provide enough information for setting up the environment?

|Name|Tested Version|
|---|---|
|Courier Prime|g20180831_1|
|Graphviz|7.1.0_7|
|Python|3.9.17|
|Sqlite3|3.42.0|

The main program to run is "chericat_graphs.py". To show all the options available and their brief description, use the --help, -h option:

```console

> $ python3.9 chericat_graphs.py --help

Reviewer: `chericat_graphs.py` should be changed to include the shebang:
```
#!/usr/bin/env python3
```
and it should have the executable permission bit set.

Then, you can simply type:
```suggestion
$ ./chericat_graphs.py --help
```

Author: Fixed the text on the README in this PR, the code changes on the .py file has been committed under a separate PR: https://github.com/CTSRD-CHERI/chericat/pull/43/commits/20764f576c9bb42a6185e9eaf834a97530a1f2d1

> usage: chericat_visualise [-h] {libview,compview} ...

Reviewer: The output says `chericat_visualise` while you execute `chericat_graphs.py`. Could it be outdated?

Author: Fixed on the text.

options:
  -h, --help          show this help message and exit

Views:
  {libview,compview}  libview or compview
    libview           Sub-commands for library-centric graphs
    compview          Sub-commands for compartment-centric graphs
```

> There are two sets of sub-commands, libveiw and compview:

Reviewer: ```suggestion
There are two sets of sub-commands, libview and compview:
```

Author: Fixed, well spotted!

```console
$ python3 chericat_graphs.py libview --help
usage: chericat_visualise libview [-h] -d D [-r R] [-g] [-c C C]

options:
  -h, --help  show this help message and exit
  -d D        The database to use for the queries
  -r R        Executes the SQL query on the provided db
  -g          Generate overview graph showing capability relationships between libraries
  -c C C      Show capabilities between two loaded libraries <libname 1> <libname 2>

$ python3 chericat_graphs.py compview --help
usage: chericat_visualise compview [-h] -d D [-r R] [-g] [-g_no_perms] [-c C C]

options:
  -h, --help   show this help message and exit
  -d D         The database to use for the queries
  -r R         Executes the SQL query on the provided db
  -g           Generate overview graph showing capability relationships between compartments
  -g_no_perms  Generate overview graph showing capability relationships (but don't show permissions) between compartments
  -c C C       Show capabilities between two compartments <compname 1> <compname 2>
```

> The output graphs are located in the graph-output directory from where you run the python program.

Reviewer: ```suggestion
The output graphs are written to the `graph-output` sub-directory in the current directory.
```

Author: Fixed, thanks for the suggestions.
